TileStache/Goodies/VecTiles/transform.py

```diff
@@ -366,3 +366,90 @@ def tags_name_i18n(shape, properties, fid, zoom):
             properties[alt_tag_name_candidate] = alt_tag_name_value
 
     return shape, properties, fid
+
+# intercut takes features from a base layer and cuts each
+# of them against a cutting layer, splitting any base
+# feature which intersects into separate inside and outside
+# parts.
+#
+# the parts of each base feature which are outside any
+# cutting feature are left unchanged. the parts which are
+# inside have their property with the key given by the
+# 'target_attribute' parameter set to the same value as the
+# property from the cutting feature with the key given by
+# the 'attribute' parameter.
+#
+# the intended use of this is to project attributes from one
+# layer to another so that they can be styled appropriately.
+#
+# returns a set of feature layers with the base layer
+# replaced by a cut one, or None if there's an error.
+def intercut(feature_layers, base_layer, cutting_layer, attribute=None, target_attribute=None):
+    base = None
+    cutting = None
+
+    # the target attribute can default to the attribute if
+    # they are distinct. but often they aren't, and that's
+    # why target_attribute is a separate parameter.
+    if target_attribute is None:
+        target_attribute = attribute
+
+    # search through all the layers and extract the ones
+    # which have the names of the base and cutting layer.
+    # it would seem to be better to use a dict() for
+    # layers, and this will give odd results if names are
+    # allowed to be duplicated.
+    for feature_layer in feature_layers:
+        layer_datum = feature_layer['layer_datum']
+        layer_name = layer_datum['name']
+
+        if layer_name == base_layer:
+            base = feature_layer
+        elif layer_name == cutting_layer:
+            cutting = feature_layer
+
+    # didn't find one or other layer - what's the appropriate
+    # thing to do here, raise an error?
+    if base is None or cutting is None:
+        return None
```

Comment:
I think an assert should be fine here, because that suggests a config problem. Maybe something like:

```python
assert base is not None and cutting is not None, 'intercut config error'
```